### PR TITLE
Order Statuses/Order Return Statuses : Use the new form theme

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Order Statuses/Order Return Statuses : Use the new form theme
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | * Go to the `/admin-dev/index.php/configure/shop/order-return-states/new` page<br>* The form theme has been updated like in #16482<br>* The Add & Edit works<br>* Go to the `/admin-dev/index.php/configure/shop/order-states/new` page<br>* The form theme has been updated like in #16482<br>* The Add & Edit works
| Fixed ticket?     | N/A
| Related PRs       | Related to #27590 
| Sponsor company   | @PrestaShopCorp

Ping @jolelievre

### CODE REVIEW

**Review by Hlavtox — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderReturnStates/OrderReturnStateType.php
Comment (FabienPapet) [inline src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderReturnStates/OrderReturnStateType.php:53]: Not sure if typo, or used like that. (applies to all usages in this file)

```suggestion
                'label' => $this->trans('Status name', 'Admin.ShopParameters.Feature'),
```

Comment (Progi1984) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderReturnStates/OrderReturnStateType.php:53]: It's not a typo. Check in `MetaController` or `AdminStatusesController`.

Comment (Hlavtox) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderReturnStates/OrderReturnStateType.php:53]: @FabienPapet Not everything is came case! :D

**Review by matthieu-rolland — DISMISSED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @Progi1984 

I have checked your PR. I think that the improvement is not added when i used these two urls : 

- ` /admin-dev/index.php/configure/shop/order-return-states/new page` : **NOK** : redirect to Dashboard page

- `/admin-dev/index.php/configure/shop/order-states/new page` : **NOK** :  redirect to Dashboard page

The new pages are not displayed!

https://user-images.githubusercontent.com/92912932/221526100-3efa75f0-9561-4331-bac6-5711253fd030.mp4


Did I miss something? Could you please check it?

Thank You!

**Comment by Progi1984** [comment]
@HanaRebaiQA Updated the "How to test ?"

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @Progi1984 

I have checked your PR. Here is the list of issues I found : 

**Order States Page :** 

**Issue 1 :** The Send of an email section in the migrated page is not coherent with the legacy page


1. `Email template` instead of `Template`
2. `Account` is the default template instead of `order_conf`
3. The language section `en - English (English)` instead of `en`
4. Preview background color `Blue` instead of `White`
5. The icon of information `i` instead of `?`
6. The design of the information section is not very much displayed: much empty space, not contoured

**Legacy :**
image

**Migrated page :** 
image

**Issue 2 :** The title in the migrated page is not the same as the legacy page

**Legacy page :** 
image

**Migrated page :** 
image

**Issue 3 :** The color section is not well displayed. Missing the field outline

**Legacy page :** 
image

**Migrated page :** 
image

**Issue 4 :**  The color palette of the migrated page is not the same as the legacy page when clicking on the color field

**Legacy page :** 
image

**Migrated page :** 
image

**Issue 5 :**  The icon field is missing in the migrated page 

**Legacy page :** 
image

**Migrated page :** 
image

**Issue 6 :** There is no default color displayed in the migrated page. the default color in the legacy page is `#ffffff`

**Legacy page :** 
image

**Migrated page :**
image

**Issue 7 :** The Email template contains new templates on the migrated page 

**Migrated page :** 
image

**Issue 8 :** The templates on the migrated page are not in the same order as the legacy page

**Legacy page:**
image

**Migrated page :**
image

**Issue 9 :** The color background when choosing any color is always white on te migrated page

**Legacy page :**
image

**Migrated page :**
image

**Issue 10 :** The help messages are displayed on the migrated page while not on the legacy one (Is it the expected behavior ?)

**Migrated page :**
image

**Issue 11 :** 403 forbidden on the legacy page when clicking on Preview button but it's OK on the migrated page (That's Cool but for the legacy page is it normal ?)

**Legacy page :**
image

**Migrated page :**
image

**Issue 12 :** An information message of the color field is displayed when putting the Name and the color empty and clicking on Save ==> Migrated page
But, an alert message is displayed instead of it on the legacy page 

**Legacy page :** 
image

**Migrated page :**
image

**Issue 13 :** The icon near the Order status title is not the same on the migrated page

**Legacy page :** 
image

**Migrated page :**
image

**Issue 14 :** These invalid characters and the numbers are permitted when using them on the name field

**Migrated page :** 

https://user-images.githubusercontent.com/92912932/222166032-97eab54a-beec-48c7-a206-e9511f7841ac.mp4

image

**Return Statuses Page :** 

**Issue 15 :** The title when adding the return statuses on the migrated page is not the same as the legacy page

**Legacy page :**
image

**Migrated page :** 
image

**Issue 16 :**  The Color field is required on the migrated page but it's not on the legacy page

**Legacy page :**
image

**Migrated page :**
image

**Issue 17 :** The title of the return statuses on the migrated page is not the same as the legacy page

**Legacy page :**
image

**Migrated page :** 
image

**Issue 18 :** The edited title of the migrated page is not the same as the legacy page (It's more but what's the expected behavior?)

**Legacy page:** 
image

**Migrated page :**
image

**Issue 19 :** An exception is thrown when editing the return order status which is created with the legacy page when the fileds are empty or only the color field is filled.

image


https://user-images.githubusercontent.com/92912932/222192568-3567e948-1a60-42ca-8903-4396b006f43d.mp4


**Issue 20 :** These issues of the Order Status page are also present on the Order return status page : idem que **Issue 3**
**Issue 21 :** idem que **Issue 4**
**Issue 22 :** idem que **Issue 6**
**Issue 23 :** idem que **Issue 9**
**Issue 24 :** idem que **Issue 10**
**Issue 25 :** idem que **Issue 13**
**Issue 26 :** idem que **Issue 14**
**Issue 27 :** idem que **Issue 12**, but with different behavior of the legacy page : 

Legacy page : When clicking on the save button and the name and the color fields are empty, there is an exception error (But it's not related to this PR, i'll investigate on it) but we should know what is the expected behavior

**Legacy page :** 
image

**Migrated page :**
image


Could you please check it ?

Thank You!

**Comment by Hlavtox** [comment]
I don't think it's needed to report things like "emails are now sorted A-Z instead of completely random order". 😄

**Review by boherm — DISMISSED** [review]
LGTM @Progi1984, anyway you can use Constructor property promotion but I not blocking this PR for this.

**Comment by Progi1984** [comment]
Hi @PrestaShop/committers, I have one approval for @boherm. Can I have a second review when you will have some availability, plz ? Thanks

**Comment by Progi1984** [comment]
@boherm Need reapprove after rebase.

**Review by boherm — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @Progi1984 ,

I have checked your PR and LGTM
Automated tests are OK

Thank you

**Comment by kpodemski** [comment]
👍🏻 🎉

**Comment by SharakPL** [comment]
Have the issues from https://github.com/PrestaShop/PrestaShop/pull/31504#pullrequestreview-1320164347 been resolved? Looks like some of that is still valid https://github.com/PrestaShop/PrestaShop/issues/35181

**Comment by Hlavtox** [comment]
@SharakPL Create an issue, nobody from this PR will answer you :D

**Pull request merged**
